Formula/mono.rb

```diff
@@ -100,32 +105,51 @@ def install
     ENV.prepend_path "PATH", bin
 
     # Next build msbuild
-    resource("msbuild").stage do
-      system "./eng/cibuild_bootstrapped_msbuild.sh", "--host_type", "mono",
-             "--configuration", "Release", "--skip_tests"
-
-      system "./stage1/mono-msbuild/msbuild", "mono/build/install.proj",
-             "/p:MonoInstallPrefix=#{prefix}", "/p:Configuration=Release-MONO",
-             "/p:IgnoreDiffFailure=true"
+    # NOTE: MSBuild 16 fails to build on Apple Silicon as it requires .NET 5
+    # TODO: MSBuild 17 seems to use .NET 6 so can try enabling when available in mono.
+    # PR ref: https://github.com/mono/msbuild/pull/435
+    unless Hardware::CPU.arm?
+      resource("msbuild").stage do
+        system "./eng/cibuild_bootstrapped_msbuild.sh", "--host_type", "mono",
```

Comment:
Was thinking that we could package `mono` without `msbuild` and use it for some dependents that only need runtime rather than build-system. Locally, I have been able to build `mono + fsharp` but not `msbuild`.

Reply:
@Homebrew/core for opinions on this and corresponding caveat. Most of Homebrew's use-case of `mono` is for runtime (usually wrapper scripts running `mono <file>.exe`). CI has successfully built ARM bottle without MSBuild.

Reply:
We can address this afterward, but I think I've been able to get `mono` to compile on Linux as well but it also fails at `msbuild`.  So we might also be able to bottle this on Linux if we skip `msbuild` there too.

Reply:
It should be possible to build `msbuild` on Linux but I don't know why we keep hitting a failure. I tried to copy ArchLinux build https://archlinux.org/packages/community/x86_64/mono-msbuild/ using other git repo, but the error is the same when getting the package index.

Upstream also provides Debian and Centos releases in the git repo https://github.com/mono/linux-packaging-msbuild/tree/main/debian but none of the build changes/patches helped.

NixOS does carry a file with all dependencies already resolved but it doesn't look practical for Homebrew.